### PR TITLE
Decompiled ovl_En_Tana and ovl_Bg_Spot05_Soko

### PR DESCRIPTION
**Pull request by shawlucas**
Also ran format.sh, explaining why a few other files were modified.

### CODE REVIEW

File: src/overlays/actors/ovl_En_Tana/z_en_tana.c [hunk 1]
```diff
@@ -20,13 +27,70 @@ const ActorInit En_Tana_InitVars = {
...
+    NULL,
+    0x06000E08,
+    0x06001608,
+};
+
+static void EnTana_Init(EnTana* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->actor;
+    osSyncPrintf("☆☆☆ %s ☆☆☆\n", shelfTypes[thisx->params]);
+    Actor_SetScale(thisx, 1.0f);
+    thisx->flags &= 0xFFFFFFFE;
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Tana/z_en_tana.c:58]:
```suggestion
    thisx->flags &= ~1;
```

File: src/overlays/actors/ovl_Bg_Spot05_Soko/z_bg_spot05_soko.c [hunk 2]
```diff
@@ -21,17 +32,77 @@ const ActorInit Bg_Spot05_Soko_InitVars = {
...
+static Gfx* dListTbl[] = {
+    0x06000840,
+    0x06001190,
+};
+
+static void BgSpot05Soko_Init(BgSpot05Soko* this, GlobalContext* globalCtx) {
+
+    Actor* thisx = &this->dyna.actor;
+    u32 sp24[2];
+    sp24[0] = 0;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Spot05_Soko/z_bg_spot05_soko.c:49]: I don't think this is supposed to be an array.

File: src/overlays/actors/ovl_Bg_Spot05_Soko/z_bg_spot05_soko.c [hunk 3]
```diff
@@ -21,17 +32,77 @@ const ActorInit Bg_Spot05_Soko_InitVars = {
...
+    thisx->speedXZ *= 1.5f;
+    if (Math_ApproxF(&thisx->posRot.pos.y, thisx->initPosRot.pos.y - 120.0f, thisx->speedXZ) != 0) {
+        Actor_Kill(thisx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot05_Soko/BgSpot05Soko_Update.s")
+static void BgSpot05Soko_Update(BgSpot05Soko* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->dyna.actor;
+    this->actionFunc(thisx, globalCtx);
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Spot05_Soko/z_bg_spot05_soko.c:102]: Action functions should normally take the specific actor struct (BgSpot05Soko in this case) not the common Actor struct, so I don't think thisx is needed at all here.

File: src/overlays/actors/ovl_Bg_Spot05_Soko/z_bg_spot05_soko.c [hunk 4]
```diff
@@ -21,17 +32,77 @@ const ActorInit Bg_Spot05_Soko_InitVars = {
...
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot05_Soko/BgSpot05Soko_Draw.s")
+extern Gfx* D_808AE724[];
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Spot05_Soko/z_bg_spot05_soko.c:105]: This doesn't seem to be used anymore?
```suggestion
```

File: src/overlays/actors/ovl_Bg_Spot05_Soko/z_bg_spot05_soko.c [hunk 5]
```diff
@@ -21,17 +32,77 @@ const ActorInit Bg_Spot05_Soko_InitVars = {
...
+            Actor_Kill(thisx);
+        } else {
+            this->actionFunc = &func_808AE5A8;
+        }
+    } else {
+        DynaPolyInfo_Alloc(&D_060012C0, &sp24);
+        if (Flags_GetSwitch(globalCtx, this->switchFlag) != 0) {
+            Actor_Kill(thisx);
+        } else {
+            this->actionFunc = &func_808AE5B4;
```
Comment (fig02) [inline src/overlays/actors/ovl_Bg_Spot05_Soko/z_bg_spot05_soko.c:66]: you dont need & for function pointers or arrays. theres a couple others in this file. other than that and romans comments, looks good

File: src/overlays/actors/ovl_Bg_Spot05_Soko/z_bg_spot05_soko.c (hunk 2 above)
Comment (shawlucas) [reply src/overlays/actors/ovl_Bg_Spot05_Soko/z_bg_spot05_soko.c:49]: If it's not an array, should I I just make it padding?

Comment (Roman971) [reply src/overlays/actors/ovl_Bg_Spot05_Soko/z_bg_spot05_soko.c:49]: If that's required for stack allocation, yes.

Comment (shawlucas) [reply src/overlays/actors/ovl_Bg_Spot05_Soko/z_bg_spot05_soko.c:49]: I tried a lot of different things and nothings matching if I don't make sp24 an array

Comment (shawlucas) [reply src/overlays/actors/ovl_Bg_Spot05_Soko/z_bg_spot05_soko.c:49]: The stack isnt even changing, even if I make the pad 5 words long. It keeps going to -0x20. But when i make sp24 an array, it matches with stack at -0x30

Comment (shawlucas) [reply src/overlays/actors/ovl_Bg_Spot05_Soko/z_bg_spot05_soko.c:49]: Oh I think I found the issue

**Comment by shawlucas** [comment]
Fixed

**Pull request merged**